lib/puma/server.rb

```diff
@@ -595,6 +600,24 @@ def handle_request(req, lines)
       env[RACK_INPUT] = body
       env[RACK_URL_SCHEME] =  env[HTTPS_KEY] ? HTTPS : HTTP
 
+      if @early_hints
+        env[EARLY_HINTS] = lambda { |links|
+          fast_write client, "HTTP/1.1 103 Early Hints\r\n".freeze
+
+          links.each do |link|
+            fast_write client, "Link: <#{link[:link]}>; rel=preload"
+            if as = link[:as]
+              fast_write client, "; as=#{as}"
```

Comment:
Should we auto-detect this part? I did some of this in [rack-http-preload.](https://github.com/nateberkopec/rack_http_preload/blob/master/lib/rack_http_preload.rb#L25).

Reply:
Unsure.  h2o pushes the resource regardless.  I think this is something we need to iterate on. (IOW, I don't have an answer)

Reply:
It's also possible we may not want to auto-detect `as` at the *puma* level. I just wanted to ask the question.